pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/DataTypeTransformer.java

```diff
@@ -109,7 +115,11 @@ public GenericRow transform(GenericRow record) {
         }
       }
       if (source != dest) {
-        value = dest.convert(value, source);
+        try {
+          value = dest.convert(value, source);
+        } catch (Exception e) {
+          throw new DataTypeConversionException(column, source, dest, e);
```

Comment:
Not really part of this PR, but figured I'd include it. This module throws errors that are pretty impossible to debug since they have no context on which column caused the explosion to happen.